File: sharetravelBackend/src/main/java/com/salesianostriana/dam/sharetravelBackend/rating/dto/NewRatingDto.java
package com.salesianostriana.dam.sharetravelBackend.rating.dto;

import jakarta.validation.constraints.*;

import java.util.UUID;

public record NewRatingDto(
        @NotNull(message = "Rating value cannot be null")
        @Min(value = 1, message = "The rating value must be at least 1")
        @Max(value = 5, message = "The rating value must be at most 5")
        @Digits(integer=10, fraction=1, message = "Rating value must not have more than one decimal")
        double ratingValue,
        @NotBlank(message = "Feedback cannot be empty")
        String feedback,
        UUID driverId
) {
}

File: sharetravel_frontend/lib/model/dto/new_rating_dto.dart
class NewRatingDto {
  double? ratingValue;
  String? feedback;
  String? driverId;

  NewRatingDto({this.ratingValue, this.feedback, this.driverId});

  NewRatingDto.fromJson(Map<String, dynamic> json) {
    ratingValue = json['ratingValue'];
    feedback = json['feedback'];
    driverId = json['driverId'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['ratingValue'] = ratingValue;
    data['feedback'] = feedback;
    data['driverId'] = driverId;
    return data;
  }
}
